src/modules/auth/auth.service.ts

```diff
@@ -53,16 +59,31 @@ export class AuthService {
       throw new BadRequestException('Invalid email or password');
     }
     const payload = { sub: user.id, userEmail: user.email };
-    const accessToken = await this.jwtService.signAsync(payload);
+    const accessToken = await this.generateToken('ACCESS', payload);
+    const refreshToken = await this.generateToken('REFRESH', payload);
 
-    return { accessToken };
+    return { accessToken, refreshToken };
   }
-  async generateAccessToken(payload: { sub: string; userEmail: string }) {
-    const accessToken = await this.jwtService.signAsync(payload, {
-      secret: this.configService.get('AUTH_ACCESS_TOKEN_SECRET'),
-      expiresIn: this.configService.get('AUTH_ACCESS_TOKEN_EXPIRES_IN'),
-    });
 
-    return accessToken;
+  async generateToken(tokenType: 'REFRESH' | 'ACCESS', payload: JwtPayload) {
+    const refreshToken = await this.jwtService.signAsync(
+      { sub: payload.sub, userEmail: payload.userEmail },
```

Comment:
I think you can also add user role here in the payload

Reply:
why? `payload.sub` (user id)is already a unique identifier

Reply:
I thought it might be needed for the authorization logic later but we are not making admin dashboard so I guess it is not needed.